python/Ganga/Utility/Shell.py

```diff
@@ -64,6 +71,69 @@ def expand_vars(env):
     return tmp_dict
 
 
+def get_env_from_arg(this_arg, def_env_on_fail = True):
+    """
+    Function to return the environment after running the command "this_command"
+    Args:
+        this_arg (str): This file/command is the command which is to be run after source to alter an environment
+        def_env_on_fail (bool): True means the default environment is to be returned on failure (i.e. I expect a dict of environ), False is return None on failure
+    """
+
+    if def_env_on_fail:
+        env = dict(os.environ)
+        env = expand_vars(env)
+    else:
+        env = None
+
+    logger.debug("Initializing Shell")
+    logger.debug("arg:\n%s" % this_arg)
+
+    this_cwd = os.path.abspath(os.getcwd())
+    if not os.path.exists(this_cwd):
+        this_cwd = os.path.abspath(tempfile.gettempdir())
+
+    logger.debug("Using CWD: %s" % this_cwd)
+    command = 'source %s > /dev/null 2>&1; python -c "from __future__ import print_function; import os; print(os.environ)"' % (this_arg)
+    logger.debug('Running:   %s' % command)
+    pipe = subprocess.Popen('bash', env=env, cwd=this_cwd, shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    output = pipe.communicate(command)
+    rc = pipe.poll()
+
+    if rc:
+        logger.warning('Unexpected rc %d from setup command %s', rc, setup)
+
+        try:
+            env2 = expand_vars(eval(eval(str(output))[0]))
```

Comment:
This is almost certainly being (ab)used by one of the plugins in the past, I remember I never understood why this worked and fixed things but I could certainly demonstrate that it did... I've rewritten this few lines to something sane now with 2 lines of docs.